engine/storage/configdrive/src/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilder.java

```diff
@@ -51,30 +47,45 @@
 
     public static final Logger LOG = Logger.getLogger(ConfigDriveBuilder.class);
 
-    private static void writeFile(final File folder, final String file, final String content) {
-        if (folder == null || Strings.isNullOrEmpty(file)) {
-            return;
-        }
-        final File vendorDataFile = new File(folder, file);
-        try (final FileWriter fw = new FileWriter(vendorDataFile); final BufferedWriter bw = new BufferedWriter(fw)) {
-            bw.write(content);
+    /**
+     * Writes a content {@link String} to a file that is going to be created in a folder. We will not append to the file if it already exists. Therefore, its content will be overwritten.
+     * Moreover, the charset used is {@link com.cloud.utils.StringUtils#getPreferredCharset()}.
+     *
+     * We expect the folder object and the file not to be null/empty.
+     */
+    static void writeFile(File folder, String file, String content) {
+        try {
+            FileUtils.write(new File(folder, file), content, com.cloud.utils.StringUtils.getPreferredCharset(), false);
         } catch (IOException ex) {
             throw new CloudRuntimeException("Failed to create config drive file " + file, ex);
         }
     }
 
-    public static String fileToBase64String(final File isoFile) throws IOException {
+    /**
+     *  Read the content of a {@link File} and convert it to a String in base 64.
+     *  We expect the content of the file to be encoded using {@link com.cloud.utils.StringUtils#getPreferredCharset()}
+     */
+    public static String fileToBase64String(File isoFile) throws IOException {
         byte[] encoded = Base64.encodeBase64(FileUtils.readFileToByteArray(isoFile));
-        return new String(encoded, StandardCharsets.US_ASCII);
+        return new String(encoded, com.cloud.utils.StringUtils.getPreferredCharset());
```

Comment:
@rafaelweingartner for base64 encoding should n't we keep it to US_ASCII?

Reply:
That is a good question. I noticed that in some places we were using US_ASCII and others the `com.cloud.utils.StringUtils.getPreferredCharset()`. I asked @DaanHoogland about it.
I can revert this change though. What do you guys think?

Reply:
I think let's stick with US_ASCII for both encoding and decoding?

Reply:
yeah, i made the same comment

Reply:
ok, so I will change that